src/app/project-sets/[projectSetId]/columns.tsx

```diff
@@ -0,0 +1,160 @@
+'use client'
+
+import {type ColumnDef} from "@tanstack/react-table"
+import {type ProjectRequirement, RequirementOperator} from "@/_temp_types/projects"
+import {DataTableColumnHeader} from "@/components/ui/data-table-column-header"
+import {
+    DropdownMenu,
+    DropdownMenuContent,
+    DropdownMenuRadioGroup,
+    DropdownMenuRadioItem,
+    DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu"
+import React from "react"
+import {ChevronDownIcon} from "@radix-ui/react-icons"
+
+
+function convertAttributeName(projectId: number): string {
+    // TODO: Get attribute name from API
+    return ""
+}
+
+export const persistedColumnDefs: ColumnDef<ProjectRequirement>[] = [
+    {
+        accessorKey: "attribute",
+        header: ({column}) => (
+            <DataTableColumnHeader column={column} title={"Attribute"}/>
+        ),
+    },
+    {
+        accessorKey: "operator",
+        header: "Operator",
+    },
+    {
+        accessorKey: "value",
+        header: "Value",
+    },
+]
+
+const allAttributes: Record<number, string> = {}
+const attributeMap: Record<number, Record<number, string>> = {}
+
+function handleUpdatedProjectRequirement(updatedRequirement: ProjectRequirement) {
+    // TODO: API call to update the project requirements
+}
+
+export const mutableColumnDefs: ColumnDef<ProjectRequirement>[] = [
+    {
+        accessorKey: "attribute",
+        header: ({column}) => (
+            <DataTableColumnHeader column={column} title={"Attribute"}/>
+        ),
+        cell: ({row}) => {
+            return (
+                <DropdownMenu>
+                    <DropdownMenuTrigger asChild>
+                        <span className="cursor-pointer flex items-center">
+                            {convertAttributeName(row.original.attribute)}
+                            <ChevronDownIcon className="ml-1 w-3 h-3"/>
+                        </span>
+                    </DropdownMenuTrigger>
+                    <DropdownMenuContent align="start">
+                        <DropdownMenuRadioGroup
+                            value={row.original.attribute.toString()}
+                            onValueChange={(value) => handleUpdatedProjectRequirement({
```

Comment:
this is kinda greedy logic, feel like it's more confusing to put this requirements code into the repo when it's so undercooked